src/components/ButtonWithModalForm/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { useTranslation } from 'react-i18next';
+import { FormProvider, useForm } from 'react-hook-form';
+import { useMutation } from '@apollo/client';
+import { useParams } from 'react-router-dom';
+import { Button } from '@mui/material';
+import { CV, USER } from '../../apollo/queries/queries.ts';
+import { PROFILE } from '../../apollo/queries/user.ts';
+import { ButtonWithModalFormProps } from '../../types/formsTypes.ts';
+import './ButtonWithModalForm.styles.scss';
+import Form from '../Form';
+import ButtonWithModal from '../ButtonWithModal';
+
+function ButtonWithModalForm<FormValues>({
+  defaultFormValues,
+  formFields,
+  modalTitle,
+  submitMutation,
+  createOpenButton,
+  transformToMutationData,
+  isUpdate,
+  readOnlyFields,
+}: ButtonWithModalFormProps<FormValues>) {
+  const [mutation] = useMutation(submitMutation);
+  const { userId } = useParams();
+
+  const methods = useForm({
+    defaultValues: defaultFormValues,
+  });
+
+  const { t } = useTranslation();
+
+  const onSubmit = (formData) => {
+    mutation({
+      variables: transformToMutationData(formData),
+      refetchQueries: [USER, PROFILE, CV],
+    });
+    if (!isUpdate) {
```

Comment:
If possible. Otherwise - leave original implementation

```suggestion
    methods.reset(!isUpdate ? formData : undefined)
```